Source/VisualStudioTools/Private/VisualStudioToolsCommandlet.cpp

```diff
@@ -99,64 +99,58 @@ struct FAssetIndex
 
 		int32 BlueprintIndex = Blueprints.Num();
 
-		bool bHasAnyParent = FindBlueprintNativeParents(BlueprintGeneratedClass, [&](UClass* Parent) {
+		bool bHasAnyParent = FindBlueprintNativeParents(BlueprintGeneratedClass, [&](UClass* Parent) 
+		{
 			FString ParentName = Parent->GetFName().ToString();
 			if (!Classes.Contains(ParentName))
 			{
 				Classes.Add(ParentName).Class = Parent;
 			}
 
-			FClassEntry& Entry = Classes[ParentName];
+			FClassEntry& ClassEntry = Classes[ParentName];
 
-			Entry.Blueprints.Add(BlueprintIndex);
+			ClassEntry.Blueprints.Add(BlueprintIndex);
 
+			// Retrieve the properties from the parent class that changed in the Blueprint class, by comparing their CDOs.
 			UObject* GeneratedClassDefault = BlueprintGeneratedClass->ClassDefaultObject;
 			UObject* SuperClassDefault = Parent->GetDefaultObject(false);
 			TArray<FProperty*> ChangedProperties = GetChangedPropertiesList(Parent, (uint8*)GeneratedClassDefault, (uint8*)SuperClassDefault);
 
 			for (FProperty* Property : ChangedProperties)
 			{
 				FString PropertyName = Property->GetFName().ToString();
-				if (!Entry.Properties.Contains(PropertyName))
+				if (!ClassEntry.Properties.Contains(PropertyName))
 				{
-					Entry.Properties.Add(PropertyName).Property = Property;
+					ClassEntry.Properties.Add(PropertyName).Property = Property;
 				}
 
-				FPropertyEntry& PropEntry = Entry.Properties[PropertyName];
+				FPropertyEntry& PropEntry = ClassEntry.Properties[PropertyName];
 				PropEntry.Blueprints.Add(BlueprintIndex);
 			}
-			});
-
-		bool bHasAnyFunctions = false;
-		for (UFunction* Fn : BlueprintGeneratedClass->CalledFunctions)
-		{
-			if (!Fn->HasAnyFunctionFlags(EFunctionFlags::FUNC_Native))
-			{
-				continue;
-			}
 
-			bHasAnyFunctions = true;
-
-			UClass* Owner = Fn->GetOwnerClass();
-			FString OwnerName = Owner->GetFName().ToString();
-			if (!Classes.Contains(OwnerName))
+			// Iterate over the functions originally from the parent class
+			// and check if they are implemented in the BP class as well.
+			for (TFieldIterator<UFunction> It(Parent, EFieldIteratorFlags::ExcludeSuper); It; ++It)
 			{
-				Classes.Add(OwnerName).Class = Owner;
-			}
+				UFunction* Fn = BlueprintGeneratedClass->FindFunctionByName((*It)->GetFName(), EIncludeSuperFlag::ExcludeSuper);
```

Comment:
The goal is to find all overridden methods, y?
Don't we need to compare also param count & param types to make this reliable?

Reply:
> The goal is to find all overridden methods, y?

The implemented methods are relevant because changing the native class can affect the blueprints that inherit from that class. E.g., a refactor that would appear safe in the C++ code would require the blueprints to be recompiled.

Ideally, we'd show _both_ overrides and usages outside of the hierarchy in the Code Lens hints, but don't because of the broad search scope of the usage case, leaving that for the Find All References feature, which is more explicit for which symbol to search.

> Don't we need to compare also param count & param types to make this reliable?

Good point, let me check that to confirm. Have not seen many cases of overloaded `UFUNCTION`s yet, so I'm not sure if it's even supported by the UE runtime.

Reply:
Validated this scenario and we seem ok: you cannot define two `UFUNCTION`s with the same name in the native class or across the hierarchy, the header tool complains it's not valid. I guess overloads are not supported by the reflection system.

As for a function in the blueprint conflicting with a function from the native parents, it does not seem to be allowed either, and the UI does not let me use the name when creating the function because it already exists.

Given that, I think the logic above should be sufficient for our needs.